config/envoyconfig/clusters.go

```diff
@@ -124,8 +124,14 @@ func (b *Builder) buildPolicyCluster(ctx context.Context, options *config.Option
 
 	cluster.AltStatName = getClusterStatsName(policy)
 
+	upstreamProtocol := upstreamProtocolAuto
+	if policy.AllowWebsockets {
+		// #2388, force http/1 when using web sockets
+		upstreamProtocol = upstreamProtocolHTTP1
```

Comment:
Can we add a log message to make clear we are using HTTP1 on this policy because websockets was set?

Reply:
To that point, I think we could abstract this little loop into a small function and call it directly in `buildPolicyEndpoints` and `buildCluster`?

Reply:
This will be very noisy, but ok.